query/src/main/java/org/infinispan/query/clustered/ClusteredCacheQueryImpl.java

```diff
@@ -39,35 +42,47 @@
    private int firstResult = 0;
 
    public ClusteredCacheQueryImpl(Query luceneQuery, SearchIntegrator searchFactory,
-            ExecutorService asyncExecutor, AdvancedCache<?, ?> cache, KeyTransformationHandler keyTransformationHandler, Class<?>... classes) {
+                                  ExecutorService asyncExecutor, AdvancedCache<?, ?> cache, KeyTransformationHandler keyTransformationHandler, Class<?>... classes) {
       super(luceneQuery, searchFactory, cache, keyTransformationHandler, null, classes);
       this.asyncExecutor = asyncExecutor;
-      this.hSearchQuery = searchFactory.createHSQuery(luceneQuery, classes);
+   }
+
+   public ClusteredCacheQueryImpl(QueryDefinition queryDefinition, ExecutorService asyncExecutor, AdvancedCache<?, ?> cache,
+                                  KeyTransformationHandler keyTransformationHandler, IndexedTypeMap<CustomTypeMetadata> metadata) {
+      super(queryDefinition, cache, keyTransformationHandler);
+      if (metadata != null) {
+         this.queryDefinition.setIndexedType(metadata.keySet().iterator().next().getPojoType());
+         this.queryDefinition.setSortableField(metadata.values().iterator().next().getSortableFields());
+      }
+      this.asyncExecutor = asyncExecutor;
    }
 
    @Override
    public CacheQuery<E> maxResults(int maxResults) {
       this.maxResults = maxResults;
+      this.queryDefinition.setMaxResults(maxResults);
       return super.maxResults(maxResults);
    }
 
    @Override
    public CacheQuery<E> firstResult(int firstResult) {
       this.firstResult = firstResult;
+      this.queryDefinition.setFirstResult(firstResult);
       return this;
    }
 
    @Override
    public CacheQuery<E> sort(Sort sort) {
       this.sort = sort;
```

Comment:
Is the ```sort``` field used at all? I see it being set but don't see it used anywhere.

Reply:
```Sort``` is used in the ctor of ```DistributedIterator```

Reply:
My bad, was looking at the wrong branch. ```sort``` is unused, I'll remove it